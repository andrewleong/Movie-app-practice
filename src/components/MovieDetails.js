import React from 'react';

const API_IMAGE = process.env.REACT_APP_API_IMAGE;

const MovieDetails = ({ movie, toggleModal }) => {
    return (
        <div className="modal-dialog" role="document">
            <div className="modal-content">
                <div className="modal-header">
                    <h5 className="modal-title" id="exampleModalLabel">Release date: {movie.release_date}</h5>
                    <button type="button" className="close" data-dismiss="modal" onClick={toggleModal} aria-label="Close">
                    <span aria-hidden="true">&times;</span>
                    </button>
                </div>
                <div className="modal-body">
                    <div className="card">
                        <img className="card-img-top" src={`${API_IMAGE}w500/${movie.poster_path}`} alt="" />
                        <div className="card-img-overlay white">
                            <h3 className="card-title">{movie.title ? movie.title : 'N/A'}</h3>
                            <h4 className="card-subtitle"><span style={{color:"#fbb829"}}>IMDB</span> ratings: {movie.vote_average}</h4>
                            <p className="card-text">{movie.overview ? movie.overview : 'N/A'}</p>
                            <p className="card-text">Duration: {movie.runtime ? `${movie.runtime} min` : 'N/A'}</p>
                            <p className="card-text">Popularity: {movie.popularity ? `${movie.popularity}` : 'N/A'}</p>
                            <p className="card-text">Budget: $ {movie.budget}</p>
                            <p className="card-text">Genre: { movie.genres ? movie.genres.map(item => <span key={item.id} className="badge badge-info">{item.name}</span>) : 'N/A' }</p>
                            <p className="card-text">Production companies: { movie.production_companies ? movie.production_companies.map(item => <span key={item.id} className="badge badge-light">{item.name}</span>) : 'N/A' }</p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default MovieDetails;
