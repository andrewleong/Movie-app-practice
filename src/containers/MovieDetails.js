import React, { Component } from 'react';

class MovieDetails extends Component {
    constructor(props) {
        super(props);
        this.state = {
            movie: {}
        }
    }


    render() {
        return (
            <div
            style={{
                display: `${this.props.showModal ? 'block' : 'none'}`,
                backgroundColor: 'rgba(0,0,0,.8)'
              }}
            >
                Modal Pop up
            </div>
        );
    }

}

export default MovieDetails;
