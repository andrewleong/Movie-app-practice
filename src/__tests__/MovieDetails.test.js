import React from 'react';
import { shallow, mount } from 'enzyme';
import MovieDetails from '../containers/MovieDetails';
import MovieDetailsComponent from '../components/MovieDetails';

const API_SITE = process.env.REACT_APP_API_SITE;
const API_KEY = process.env.REACT_APP_API_KEY;


describe('(Component) MovieDetails', () => {
    it('should render a modal of movie details correctly', () => {
        const movie = {
            title: 'movie 1',
            id: 122,
            release_date: '12-12-2019'
        }
        const wrapper = mount(
            <MovieDetailsComponent
                movie={movie}
            />
        );

        expect(wrapper.props()).toEqual({movie});
        expect(wrapper.props()).toMatchSnapshot();
    })
});

describe('(Container) MovieDetails', () => {
    it('should fetch movie details when rendered', () => {
        const fetchSpy = jest.spyOn(window, 'fetch');
        const wrapper = shallow(
            <MovieDetails movieDetails={{ id: 64330 }} />
        );
        expect(fetchSpy).toBeCalled();
        expect(fetchSpy).toBeCalledWith(`${API_SITE}movie/64330?api_key=${API_KEY}`);
    });
});
