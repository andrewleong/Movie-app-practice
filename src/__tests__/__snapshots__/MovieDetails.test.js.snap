// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) MovieDetails should render a modal of movie details correctly 1`] = `
Object {
  "movie": Object {
    "id": 122,
    "release_date": "12-12-2019",
    "title": "movie 1",
  },
}
`;
